app/server.go

```diff
@@ -2060,12 +2060,12 @@ func withMut(mut *sync.Mutex, f func()) {
 	f()
 }
 
-func cancelTask(mut *sync.Mutex, task *model.ScheduledTask) {
+func cancelTask(mut *sync.Mutex, task **model.ScheduledTask) {
 	mut.Lock()
 	defer mut.Unlock()
-	if task != nil {
-		task.Cancel()
-		task = nil
```

Comment:
Here just the local variable (`task`) containing the pointer was being set to `nil`. 
We needed to set the pointer of the global task variable (`a.ch.dndTask` or `a.ch.postReminderTask`, depending on where this function was called) to nil.